src/Symfony/Component/HttpClient/HttpClient.php

```diff
@@ -30,7 +31,7 @@ final class HttpClient
      */
     public static function create(array $defaultOptions = [], int $maxHostConnections = 6, int $maxPendingPushes = 50): HttpClientInterface
     {
-        if ($amp = class_exists(ConnectionLimitingPool::class)) {
+        if ($amp = class_exists(ConnectionLimitingPool::class) && interface_exists(Promise::class)) {
```

Comment:
Alternatively, we could check 1 class/interface that exists in v4 and not in v5 here instead of having 2 checks. But I followed @kelunik (maintainer of amphp) suggestion of using this core amphp interface (http client v4 requires core v2 that has this interface, http client v5 requires core v3 that doesn't have this interface).